### ISSUE TITLE
arlo integration won't connect since today. 2fa error

### ISSUE THREAD

**Issue by platini76** [twrecked/hass-aarlo#667]
my config:

aarlo:
  username: correct
  password: correct
  refresh_devices_every: 2
  stream_timeout: 120
  reconnect_every: 90
  backend: sse
  tfa_source: imap
  tfa_type: email
  tfa_host: imap.mail.me.com
  tfa_username: correct
  tfa_password: correct 
  user_agent: Chrome
  stream_snapshot: True
  verbose_debug: True

credential are same of yesterday.. but today won't work....

someone can help me?

**Comment by pcridland** [comment]
Same thing happening on mine. Below is the error from core log

2022-11-24 10:15:24.247 ERROR (SyncWorker_6) [pyaarlo] imap connection failed[SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:997)
2022-11-24 10:20:25.620 ERROR (SyncWorker_6) [pyaarlo] 2fa startup failed

Config:
aarlo:
  host: https://my.arlo.com
  username: !secret arlo_user
  password: !secret arlo_pass
  db_motion_time: 30
  db_ding_time: 10
  recent_time: 10
  last_format: '%m-%d %H:%M'
  refresh_devices_every: 2
  verbose_debug: True
  tfa_source: imap
  tfa_type: email
  tfa_host: 'imap.shaw.ca'
  tfa_username: !secret shaw_user
  tfa_password: !secret shaw_pass

**Comment by twrecked** [comment]
I'll have to look; the IMAP client just uses the system certs but I'll see if I can change something.

**Comment by platini76** [comment]
yes I keep system updated...

**Comment by pcridland** [comment]
Yes I upgrade whenever it comes available as well. I thought it might be something with the certs but duckDNS and external access still works fine through HTTPS

**Comment by platini76** [comment]
I have no https.. only duckdns

**Comment by twrecked** [comment]
Can you try adding this to the _Aarlo_ config:

```
aarlo:
  ...
  default_ciphers: True
```

And I think it's to do with the server sending back a cert we think is too weak. I guess they changed the settings in the _Home Assistant_  image to prefer stronger encryption.

**Comment by platini76** [comment]
> Can you try adding this to the _Aarlo_ config:
> 
> ```
> aarlo:
>   ...
>   default_ciphers: True
> ```
> 
> And I think it's to do with the server sending back a cert we think is too weak. I guess they changed the settings in the _Home Assistant_ image to prefer stronger encryption.

sorry it will not solve... nothing change..

**Comment by twrecked** [comment]
It looks like Python tightened up the cipher requirements.  Other integrations are having issues.

https://github.com/home-assistant/core/issues/74551

I'll see what I can do at our end to lower the requirements for IMAP connections.

**Comment by twrecked** [comment]
Can you try the latest it allows you to select a custom cipher list. Looking online this configuration change might help (I can't really test this here, my connections are still working on the latest release).

```yaml
aarlo:
  ...
  cipher_list: "HIGH:!DH:!aNULL"
```

**Comment by pcridland** [comment]
Worked for me! Thanks for the hard work!

**Comment by platini76** [comment]
it gave me an error:


aarlo:
  username: platini76@me.com
  password: Portoric0@
  refresh_devices_every: 2
  stream_timeout: 120
  reconnect_every: 90
  backend: sse
  tfa_source: imap
  tfa_type: email
  tfa_host: imap.mail.me.com:993
  tfa_username: info@dvd24h.it
  tfa_password: hsxd-qyqt-ssda-fqku
  user_agent: Chrome
  cipher_list: "HIGH:!DH:!aNULL"
  stream_snapshot: True
  verbose_debug: True

**Comment by twrecked** [comment]
Be careful, you just posted your passwords which is why I deleted the comment. You might want to use a secrets file to make it easier to paste your config.

Did you update your `hass-aarlo` version? You need to be on the latest version for `cipher_list` to work.

**Comment by platini76** [comment]
ups sorry....

**Comment by twrecked** [comment]
No worries. Did it work in the end?

**Comment by platini76** [comment]
no, email did not arrive... it seems to keep oldest settings...

**Comment by twrecked** [comment]
You can check, turn on some debug and look for the word `custom`, you should see the cipher list you configured.

**Comment by platini76** [comment]
2022-11-27 18:24:02.962 WARNING (MainThread) [homeassistant.setup] Setup of aarlo is taking over 10 seconds.
2022-11-27 18:24:52.135 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: aarlo
2022-11-27 18:24:56.428 ERROR (SyncWorker_5) [pyaarlo] 2fa core retrieval failed
2022-11-27 18:24:56.449 ERROR (SyncWorker_5) [custom_components.aarlo] unable to connect to Arlo: attempt=1,sleep=15,error=2fa core retrieval failed
2022-11-27 18:25:52.257 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: aarlo
2022-11-27 18:26:14.123 ERROR (SyncWorker_5) [pyaarlo] 2fa core retrieval failed
2022-11-27 18:26:14.138 ERROR (SyncWorker_5) [custom_components.aarlo] unable to connect to Arlo: attempt=2,sleep=30,error=2fa core retrieval failed
2022-11-27 18:26:52.340 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: aarlo
2022-11-27 18:26:56.262 WARNING (MainThread) [homeassistant.components.template.config] The entity definition format under template: differs from the platform configuration format. See https://www.home-assistant.io/integrations/template#configuration-for-trigger-based-template-sensors
2022-11-27 18:27:04.460 WARNING (MainThread) [homeassistant.components.template.config] The entity definition format under template: differs from the platform configuration format. See https://www.home-assistant.io/integrations/template#configuration-for-trigger-based-template-sensors
2022-11-27 18:27:07.122 WARNING (MainThread) [homeassistant.components.template.config] The entity definition format under template: differs from the platform configuration format. See https://www.home-assistant.io/integrations/template#configuration-for-trigger-based-template-sensors
2022-11-27 18:27:47.150 ERROR (SyncWorker_5) [pyaarlo] 2fa core retrieval failed
2022-11-27 18:27:47.164 ERROR (SyncWorker_5) [custom_components.aarlo] unable to connect to Arlo: attempt=3,sleep=60,error=2fa core retrieval failed
2022-11-27 18:27:52.467 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: aarlo
2022-11-27 18:28:52.565 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: aarlo
2022-11-27 18:28:52.962 ERROR (MainThread) [homeassistant.setup] Setup of aarlo is taking longer than 300 seconds. Startup will proceed without waiting any longer
2022-11-27 18:28:52.966 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.binary_sensor: Unable to set up component.
2022-11-27 18:28:52.974 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.sensor: Unable to set up component.
2022-11-27 18:28:52.978 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.alarm_control_panel: Unable to set up component.
2022-11-27 18:28:52.980 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.light: Unable to set up component.
2022-11-27 18:28:52.983 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.media_player: Unable to set up component.
2022-11-27 18:28:52.986 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.camera: Unable to set up component.
2022-11-27 18:28:52.988 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform aarlo.switch: Unable to set up component.
2022-11-27 18:28:53.292 WARNING (MainThread) [homeassistant.components.netatmo] Webhook not registered - https and port 443 is required to register the webhook
2022-11-27 18:28:53.417 ERROR (MainThread) [homeassistant.config_entries] Cannot find integration matter
2022-11-27 18:28:53.462 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved

**Comment by platini76** [comment]
2022-11-27 18:49:29.135 DEBUG (SyncWorker_3) [pyaarlo] request-end=200
2022-11-27 18:49:29.135 DEBUG (SyncWorker_3) [pyaarlo] looking for email
2022-11-27 18:49:29.136 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: starting
2022-11-27 18:49:29.159 DEBUG (SyncWorker_3) [pyaarlo] imap is using custom ciphers HIGH:!DH:!aNULL
2022-11-27 18:49:30.785 DEBUG (SyncWorker_3) [pyaarlo] old-ids=[b'']
2022-11-27 18:49:30.785 DEBUG (SyncWorker_3) [pyaarlo] starting auth with email
2022-11-27 18:49:30.788 DEBUG (SyncWorker_3) [pyaarlo] request-url=https://ocapi-app.arlo.com/api/startAuth?event_id=FE!edd49fe8-8f6c-4c9b-9062-d6b1348f851b&time=1669571370788
2022-11-27 18:49:30.789 DEBUG (SyncWorker_3) [pyaarlo] request-params=
{'factorId': 'UGluZ0lkU0RLOkVNQUlMOjA2ODxxxxxxtNGZjNy0wNjg2LTQ3NTI1ZjJjNGZjNzo1NGFhMDk5ZS0zNjZmLTQ3YjUtOTM5YS1iZmY3MTNlMWJkOTY='}
2022-11-27 18:49:30.790 DEBUG (SyncWorker_3) [pyaarlo] request-headers=
{'Accept': 'application/json, text/plain, */*',
 'Accept-Language': 'en-US,en;q=0.9',
 'Authorization': 'Ml9veHNxdjIybkNzdnpXZ1VJclZhRXBoZxxxxxxxxxxxxRKWnFfNW9IdUcwM2JWTDRWb0ZETFVVcmwxMEwzN091dTJQN1hqV3Z3SnVGS1RicS1LeWdBb3ZVVDgtcEpObDB4Z1VIUTRkS2tWY011TG5JX3R1TUhTdklmMm5HWjFMME1DVmFnVmNQRzU1bWVjdWdpSTlKYmdCUkl6aTdvS2dnX21aLXJiMnpwRjNjcFZGVXhWaXFmM0YxRE5sVDkzVEk1S3hTa3ZCNE5kckVOcjZoN29zUmYxUS0ycm9VeHhweDA5N01tbUI3czRGTmduRlREdExPa3VFcGd6T3dwbklvckFoTnp2QUw3c2M5eUhWWktjaUFoNWNEYXZjbXBqTS1YNlFoZFF3cnl3TzkwWng4bEZXUC04UXRSYWpqdUh6bjdsa0dPZEVzVWhWQXJnYUQ1angx',
 'Origin': 'https://my.arlo.com',
 'Referer': 'https://my.arlo.com/',
 'Source': 'arloCamWeb',
 'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, '
               'like Gecko) Chrome/88.0.4324.96 Safari/537.36'}
2022-11-27 18:49:31.411 DEBUG (SyncWorker_3) [pyaarlo] request-body=
{'data': {'MFA_State': 'ENABLED',
          '_type': 'FactorAuthCode',
          'factorAuthCode': 'UGluZ0lkU0RLOkVNQUlMOndlYnNxxxxxx00NGM2LWE2NDItYWNhYTQ0NTg0NzlhOjU0YWEwOTllLTM2NmYtNDdiNS05MzlhLWJmZjcxM2UxYmQ5Ng=='},
 'meta': {'code': 200}}
2022-11-27 18:49:31.411 DEBUG (SyncWorker_3) [pyaarlo] request-end=200
2022-11-27 18:49:31.411 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: checking
2022-11-27 18:49:34.774 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: new-ids=[b'']
2022-11-27 18:49:34.775 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: no change in emails
2022-11-27 18:49:38.138 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: new-ids=[b'']
2022-11-27 18:49:38.139 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: no change in emails
2022-11-27 18:49:41.500 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: new-ids=[b'']
2022-11-27 18:49:41.501 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: no change in emails

**Comment by twrecked** [comment]
It's logging in but not seeing the email.

Is it going to the spam folder?  Can you see the email at all in them email account?

**Comment by platini76** [comment]
no,now email arrive... many emails...

**Comment by twrecked** [comment]
The problem is _Aarlo_ isn't seeing them.

```
2022-11-27 18:49:34.774 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: new-ids=[b'']
2022-11-27 18:49:34.775 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: no change in emails
2022-11-27 18:49:38.138 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: new-ids=[b'']
2022-11-27 18:49:38.139 DEBUG (SyncWorker_3) [pyaarlo] 2fa-imap: no change in emails
```

If we were we'd see numbers appearing in the `new-ids` line.

Can you tell me what the `From:` field in the email from _Arlo_ is. The code is expecting `do_not_reply@arlo.com`.

**Comment by platini76** [comment]
Received: from st11p00im-ztav06212001.me.com by p41-mailgateway-smtp-fc975c9b5-dlxqz (mailgateway 2302B297)
	with SMTP id bf6f9e95-ba11-4204-86f6-50c3d5258691 
	for <platini76@me.com>; Sun, 27 Nov 2022 20:39:20 GMT
Received: from e239-156.smtp-out.eu-west-1.amazonses.com (e239-156.smtp-out.eu-west-1.amazonses.com [23.251.239.156])
	by st11p00im-ztav06212001.me.com (Postfix) with ESMTPS id 319E6242FAF
	for <platini76@me.com>; Sun, 27 Nov 2022 20:39:18 +0000 (UTC)
From: Arlo <do_not_reply@arlo.com>
To: "platini76@me.com" <platini76@me.com>
Subject: Your one-time authentication code from Arlo
Thread-Topic: Your one-time authentication code from Arlo
Thread-Index: ATAyLWIwK+MTWkKBp9XgmJ/4IxSlgw==
X-MS-Exchange-MessageSentRepresentingType: 1
Date: Sun, 27 Nov 2022 20:39:15 +0000
Message-ID:
	<01020184bad0c7ca-db326fbc-e77d-49bf-ba13-db19b12a30fd-000000@eu-west-1.amazonses.com>
Reply-To: Arlo <do_not_reply@arlo.com>
Content-Language: en-US
X-MS-Has-Attach:
X-MS-Exchange-Organization-SCL: -1
X-MS-TNEF-Correlator:
X-MS-Exchange-Organization-RecordReviewCfmType: 0
received-spf: pass (spf.icloud.com: domain of
 01020184bad0c7ca-db326fbc-e77d-49bf-ba13-db19b12a30fd-000000@eu-west-1.amazonses.com
 designates 23.251.239.156 as permitted sender) receiver=spf.icloud.com;
 client-ip=23.251.239.156; helo=e239-156.smtp-out.eu-west-1.amazonses.com;
 envelope-from=01020184bad0c7ca-db326fbc-e77d-49bf-ba13-db19b12a30fd-000000@eu-west-1.amazonses.com
Content-Type: multipart/alternative;
	boundary="_000_01020184bad0c7cadb326fbce77d49bfba13db19b12a30fd000000e_"
MIME-Version: 1.0

**Comment by twrecked** [comment]
From the config you posted earlier, is your TFA username in the configuration correct? It was different than the one in the email.

**Comment by platini76** [comment]
I changed it.. but now email arrived..

**Comment by platini76** [comment]
seems now is working.. after 9 or 10 reboots... without no changes..!!!!

but with this bug..
https://github.com/twrecked/hass-aarlo/issues/665
